STM32Cube/Tasks/trajectory.c

```diff
@@ -0,0 +1,256 @@
+}
+
+/**@return drag force acting on rocket
+ * @param extension of air brakes, used for adjusting rocket area and iterpolating Ansys sims (0-1)
+ * @param velocity used to lookup drag force from Ansys sims
+ * @param altitude used to adjust fluid density since Ansys sims were calculated at 1000m
+ * */
+float interpolate_drag(float extension, float velocity, float altitude) {
+    float drag;
+    float ext_diff = 0.5;
+    float drag_low; //drag at low ext
+    float drag_high; //drag at high ext
+    float ext_low; //low ext
+
+    if(extension < 0){
+        extension = 0;
+    } else if(extension > 1){
+        extension = 1;
+    }
+
+    if (extension > 0.5) {
+        ext_low = 0.5;
+        drag_low = lookup_drag(0.5, velocity);
+        drag_high = lookup_drag(1, velocity);
+    }
+    else { // extension < 0.5
+        ext_low = 0;
+        drag_low = lookup_drag(0, velocity);
+        drag_high = lookup_drag(0.5, velocity);
+    }
+
+    drag = drag_low + (drag_high-drag_low) / (ext_diff) * (extension - ext_low);
+    drag = drag * air_density(altitude) * rocket_area(extension); //correct the drag using the actual airbrake extension area and air density
+    return drag;
+}
+
+/**
+* @return Cd value corresponding to interpolated drag force
+*/
+float interpolate_cd(float extension, float velocity, float altitude){
+    float drag_force = interpolate_drag(extension, velocity, altitude);
+    return 2 * drag_force / (air_density(altitude) * rocket_area(extension) * (velocity * velocity)); //Cd
+}
+
+/**
+    * @param airbrake_ext extension of airbrakes, 0-1
+    * @param mass of rocket (kg)
+    * @param velX, velocity in X direction (m/s)
+    * @param velY, velocity in Y direction (m/s)
+    * @param alt, altitude (m)
+    * @return forces acting on rocket in the X and Y directions (N)
+*/
+Forces get_forces(float extension, float mass, float velX, float velY, float alt){
+    Forces forces;
+    float angle = atan(velX/velY);
+    float Fd = -interpolate_drag(extension, sqrt(velY*velY+ velX*velX), alt); // force of drag (N)
+    float Fg = -gravitational_acceleration(alt) * mass; // force of gravity (N)
+    forces.Fy = Fd * cos(angle) + Fg;
+    forces.Fx = Fd * sin(angle);
+    return forces;
+}
+
+/**
+    * rk4 method to integrate altitude from velocity, and integrate velocity from acceleration (force/mass)
+    * @param h time step
+    * @param extension extension of airbrakes, 0-1
+    * @param mass of rocket (kg)
+    * @param state, including altitude (m) and velocity in X and Y directions (m/s)
+    * @return updated altitude and velocity integrals after one rk4 step
+*/
+RK4State rk4(float h, float mass, float extension, RK4State state) {
+    Forces forces;
+
+    forces = get_forces(extension, mass, state.velX, state.velY, state.alt);
+    float ka1 = h * state.velY;
+    float kvY1 = h * acceleration(forces.Fy, mass);
```

Comment:
imo more clear to not have accel as a function and do the division, considering the variables are named force and mass, it should be simpler, we can add a comment saying accel